### PR TITLE
PIN-4923: generate presigned url

### PR DESCRIPTION
**Pull request by AsterITA**
Tested with the following successes:

Screenshot 2024-06-04 alle 16 08 08

### CODE REVIEW

File: file-manager/src/main/scala/it/pagopa/interop/commons/files/service/impl/S3ManagerImpl.scala [hunk 1]
```diff
@@ -162,4 +166,23 @@ final class S3ManagerImpl(blockingExecutionContext: ExecutionContextExecutor)(
   def calcContentMd5(file: File): String = calcContentMd5(Files.readAllBytes(file.toPath))
 
   def calcContentMd5(byteArray: Array[Byte]): String = new String(Base64.encodeBase64(DigestUtils.md5(byteArray)))
+
+  override def generatePresignedUrl(containerPath: String, path: String): Try[String] = {
```
Comment (galales) [inline file-manager/src/main/scala/it/pagopa/interop/commons/files/service/impl/S3ManagerImpl.scala]: this implementation is ok, but it creates a presigned url only for retrieving an object. It is still required, but for this specific ticket you need an presigned url to store an object.
See [this](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/examples-s3-presign.html#put-presignedobject) doc.

I suggest to rename the function to make it explicit
```suggestion
  override def generateGetPresignedUrl(containerPath: String, path: String): Try[String] = {
```

File: file-manager/src/main/scala/it/pagopa/interop/commons/files/service/impl/S3ManagerImpl.scala [hunk 2]
```diff
@@ -162,4 +166,23 @@ final class S3ManagerImpl(blockingExecutionContext: ExecutionContextExecutor)(
...
+    Using(S3Presigner.create()) { s3Presigner =>
+      val getObjectRequest = GetObjectRequest
+        .builder()
+        .bucket(containerPath)
+        .key(path)
+        .build()
+
+      val presignRequest = GetObjectPresignRequest
+        .builder()
+        .signatureDuration(Duration.ofMinutes(1))
```
Comment (galales) [inline file-manager/src/main/scala/it/pagopa/interop/commons/files/service/impl/S3ManagerImpl.scala]: may you turn this value into a configuration from env var?
1 minute may be a bit low (I suggest at least 5), anyway it's ok as default value, until we are able to override it with the configs

**Review by galales — CHANGES_REQUESTED** [review]

File: file-manager/src/main/resources/application.conf [hunk 3]
```diff
@@ -0,0 +1,2 @@
+get-url-duration-minutes = ${GET_URL_DURATION_MINUTES}
+put-url-duration-minutes = ${PUT_URL_DURATION_MINUTES}
```
Comment (galales) [inline file-manager/src/main/resources/application.conf]:
```suggestion
get-url-duration-minutes = ${PRESIGNED_URL_GET_DURATION_MINUTES}
put-url-duration-minutes = ${PRESIGNED_URL_PUT_DURATION_MINUTES}
```

**Review by galales — APPROVED** [review]
very minor request change, but LGTM!

File: file-manager/src/main/scala/it/pagopa/interop/commons/files/StorageConfiguration.scala [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
-import com.typesafe.config.ConfigFactory
+import com.typesafe.config.{Config, ConfigFactory}
 
 object StorageConfiguration {
-  val maxConcurrency: Int = ConfigFactory.load().getInt("interop-commons.storage.max-concurrency")
+  val config: Config = ConfigFactory.load()
+
+  val maxConcurrency: Int        = config.getInt("interop-commons.storage.max-concurrency")
+  val getUrlDurationMinutes: Int = config.getInt("get-url-duration-minutes")
+  val putUrlDurationMinutes: Int = config.getInt("put-url-duration-minutes")
```
Comment (beetlecrunch) [inline file-manager/src/main/scala/it/pagopa/interop/commons/files/StorageConfiguration.scala]: These new configurations mean we should add the new environment variables everywhere we use the FileManager.
We can avoid this by not updating the Commons library version except in BFF.

**Review by beetlecrunch — APPROVED** [review]

File: file-manager/src/main/scala/it/pagopa/interop/commons/files/StorageConfiguration.scala (hunk 4 above)
Comment (galales) [reply file-manager/src/main/scala/it/pagopa/interop/commons/files/StorageConfiguration.scala]: mmm it's a very good point, it's not ideal that one service must declare them even when not using the feature. My bad.
What if they become function parameters and we set them directly in the service (BFF)?

Comment (beetlecrunch) [reply file-manager/src/main/scala/it/pagopa/interop/commons/files/StorageConfiguration.scala]: It seems like a more than reasonable solution to me.

File: file-manager/src/main/scala/it/pagopa/interop/commons/files/service/FileManager.scala [hunk 5]
```diff
@@ -42,9 +42,9 @@ trait FileManager {
 
   def close(): Unit
 
-  def generateGetPresignedUrl(bucketName: String, path: String, fileName: String): Try[String]
+  def generateGetPresignedUrl(bucketName: String, path: String, fileName: String, duration: Int): Try[String]
```
Comment (galales) [inline file-manager/src/main/scala/it/pagopa/interop/commons/files/service/FileManager.scala]: `duration` should be of type `FiniteDuration`, or leaving as `Int` is ok, but I suggest renaming it to `durationInMinutes`

File: file-manager/src/main/scala/it/pagopa/interop/commons/files/service/FileManager.scala [hunk 6]
```diff
@@ -42,9 +42,9 @@ trait FileManager {
...
 
-  def generatePutPresignedUrl(bucketName: String, path: String, fileName: String): Try[String]
+  def generatePutPresignedUrl(bucketName: String, path: String, fileName: String, duration: Int): Try[String]
```
Comment (galales) [inline file-manager/src/main/scala/it/pagopa/interop/commons/files/service/FileManager.scala]: same

**Review by galales — APPROVED** [review]

**Pull request merged**